sycl/include/CL/sycl/enums.hpp

```diff
@@ -0,0 +1,29 @@
+//==-------------- enums.hpp --- SYCL enums --------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+enum class memory_scope : unsigned char {
+  work_item = 0,
+  sub_group = 1,
+  work_group = 2,
+  device = 3,
+  system = 4
```

Comment:
nit:
```suggestion
  work_item = 0,
  sub_group,
  work_group,
  device,
  system
```

Reply:
I'd like to keep all the numbers explicit in case we decide to add something else here.